app/models/notifications/mentioned.rb

```diff
@@ -13,19 +13,26 @@ def deleted_translation_key
     end
 
     def linked_object
-      target.post
+      target.mentions_container
     end
 
     def self.notify(mentionable, recipient_user_ids)
       actor = mentionable.author
+      relevant_mentions = filter_mentions(
+        mentionable.mentions.local.where.not(person: mentionable.author),
+        mentionable,
+        recipient_user_ids
+      )
 
-      mentionable.mentions.select {|mention| mention.person.local? }.each do |mention|
+      relevant_mentions.each do |mention|
         recipient = mention.person
-
-        next if recipient == actor || !(mentionable.public || recipient_user_ids.include?(recipient.owner_id))
-
         create_notification(recipient.owner_id, mention, actor).email_the_user(mention, actor)
       end
     end
+
+    def self.filter_mentions(mentions, mentionable, recipient_user_ids)
```

Comment:
So the point where I don't like the inheritance here is, that the method on the parent class is the "special case" (filter on private posts), and the same method on the child class, should be the default case (just no filtering). But invert the inheritance (extend `Mentioned` from `MentionedInComment`) feels also wrong.

So maybe the solution is to add a `AbstractMentioned` with the default (no filtering) `filter_mentions`, `notify` and `linked_object` method, and then both of `Mentioned` and `MentionedInComment` extending from this, and add the translation_keys and override the `filter_mentions` in the `Mentioned` class.

Just some thoughts ...